benchmarks/gdscript/spectral_norm.gd

```diff
@@ -9,52 +9,55 @@ func eval_a(i: int, j: int) -> float:
 	return 1.0 / (ij * (ij - 1) * 0.5 + i)
 
 
-func av(x, y, n: int) -> void:
+func av(x: PackedFloat64Array, y: PackedFloat64Array, n: int) -> void:
 	for i in n:
 		var a := 0.0
 		for j in n:
 			a = a + x[j] * eval_a(i, j)
 		y[i] = a
 
 
-func atv(x, y, n: int) -> void:
+func atv(x: PackedFloat64Array, y: PackedFloat64Array, n: int) -> void:
 	for i in n:
 		var a := 0.0
 		for j in n:
 			a = a + x[j] * eval_a(j, i)
 		y[i] = a
 
 
-func at_av(x, y, t, n: int) -> void:
+func at_av(x: PackedFloat64Array, y: PackedFloat64Array, t: PackedFloat64Array, n: int) -> void:
 	av(x, t, n)
 	atv(t, y, n)
 
 
 func calculate_spectral_norm(n: int) -> void:
-	var u := {}
-	var v := {}
-	var t := {}
+	var u := PackedFloat64Array([])
+	u.resize(n)
+	var v := PackedFloat64Array([])
+	v.resize(n)
+	var t := PackedFloat64Array([])
+	t.resize(n)
 	for i in n:
-		u[i] = 1
+		u[i] = 1.0
```

Comment:
I think you can use `fill()` method here to immediately fill the array with the right value. I don't remember if it exists with packed arrays though.

Reply:
It does indeed exist, I updated the code.